### ISSUE TITLE
[Grid] scrolling vertically causes unexpected behavior

### ISSUE THREAD

**Issue by svetq** [telerik/kendo-angular#2342]
**Describe the bug**
When using a navigable Grid with virtual scrolling or virtual columns and the active cell of the Grid is not visible (the Grid is scrolled horizontally) then on scrolling vertically the Grid will move automatically the horizontal scrollbar.

**To Reproduce**
Check the following example:
Virtual scrolling - https://stackblitz.com/edit/angular-4hs2lm?file=app/app.component.ts
Virtual columns - https://stackblitz.com/edit/angular-4fscm7?file=src%2Fapp%2Fapp.component.ts

1. Click a cell from the first "Product Name" column in order to focus it.
2. Scroll horizontally to the last column.
3. Scroll vertically.
![chrome_dsvszWaXYm](https://user-images.githubusercontent.com/49076727/189300766-a41f1d90-6f1c-4b34-8b28-947df0a9a265.gif)

**Workaround**

**Browser**
 - OS: [for example, iOS]
 - Browser [for example, Chrome, Safari]
 - Version [for example, 22]

**Additional context**

**Comment by Raisolution** [comment]
Also reported in thread 1555403

**Comment by mclear01** [comment]
Facing similar issue.  Any update on when this issue may be addressed?

**Comment by naveedsheikh2304** [comment]
Facing similar issue for a possible another scenario i.e. when adding a column dynamically to the grid the horizontal scrollbar shifts towards right when navigable set to true or even false.

**Comment by shivambindal93** [comment]
+1

**Comment by Kir4n** [comment]
Work around capture the contentScroll event and move the focus out of the grid to a dummy element outside the grid.

https://angular-4fscm7-my8kfm.stackblitz.io

**Comment by ankurgupta12** [comment]
> Work around capture the contentScroll event and move the focus out of the grid to a dummy element outside the grid.
> 
> https://angular-4fscm7-my8kfm.stackblitz.io



I have checked this solution but still not working for me.

**Comment by Kir4n** [comment]
Worst case comment out the cursor change event subscription altogether till it gets fixed by Kendo team. Setting Navigable to false also will do but then you will loose any navigability.

Find the following line in node_modules\@progress\kendo-angular-grid\fesm2015\kendo-angular-grid.js
```
this.subs.add(this.cursor.changes
            .subscribe(args => this.onCursorChanges(args)));
```
Comment it out.

Please not that angular cache needs to be cleaned for the change to take effect. You can verify this by inspecting the Kendo grid source in browser.

**Comment by shivambindal93** [comment]
> Worst case comment out the cursor change event subscription altogether till it gets fixed by Kendo team. Setting Navigable to false also will do but then you will loose any navigability.
> 
> Find the following line in node_modules@progress\kendo-angular-grid\fesm2015\kendo-angular-grid.js
> 
> ```
> this.subs.add(this.cursor.changes
>             .subscribe(args => this.onCursorChanges(args)));
> ```
> 
> Comment it out.
> 
> Please not that angular cache needs to be cleaned for the change to take effect. You can verify this by inspecting the Kendo grid source in browser.

@Kir4n tried updating  the function definition in the ngAfterViewInit hook, the fix initially works when the virtualColumns property is marked as "true", but has side effects, where navigation sometimes fails. Adding the code below and stack blitz link for same

code link ->https://stackblitz.com/edit/angular-4fscm7-vjak6f?file=src/app/app.component.ts
application link ->https://angular-4fscm7-vjak6f.stackblitz.io
 if (this.kendoGrid && this.kendoGrid.navigationService) {
      this.kendoGrid.navigationService.onCursorChanges = () => {};
    }

**Comment by manoj-bheemireddy** [comment]
any solution for this issue?

**Comment by slavenai** [comment]
Fixed in `v15.4.0` of `kendo-angular-grid` package.
